### PR TITLE
Networking with URLSession

### LINKED ISSUE #16863: Mobile IAP Package: Create Networking stack
Project Issue: https://github.com/wordpress-mobile/WordPress-iOS/issues/16861

## Description

- [x] Pass bearer token from the app to authenticate
- [x] Endpoint 1: Fetch SKUs
- [x] Endpoint 2: Purchase product with receipt

### PR DESCRIPTION
**Pull request by momo-ozawa**
## Description

Resolves https://github.com/wordpress-mobile/WordPress-iOS/issues/16863

This PR adds the initial networking layer:

- Added `InAppPurchasesService` to call the fetch product skus / create order endpoints
- Added initial `InAppPurchasesServiceTests`
- Added `InAppPurchasesAPIRouter` to organize the endpoints 

Endpoints ref: pdekpA-18-p2

## Notes
- This PR is a combination of two earlier PRs
    - https://github.com/Automattic/MobilePay-Apple/pull/5 (approved)
    - https://github.com/Automattic/MobilePay-Apple/pull/7 

## How to test 

1. Run the iOS app
2. Tap on Rocket Fuel to purchase
3. Tap Confirm
4. Tap OK
5. On Xcode debugger menu, click on the Manage StoreKit Transactions menu item 
6. ✅ There should be a record of Rocket Fuel being purchased (See example below)

Screen Shot 2021-07-19 at 16 32 32

### CODE REVIEW

File: Source/Services/InAppPurchasesService.swift [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+//            .decode(type: [String].self, decoder: JSONDecoder())
+//            .eraseToAnyPublisher()
+
+        let productIdentifiers = [
+            "com.mobilepay.consumable.rocketfuel",
+            "com.mobilepay.consumable.premiumrocketfuel"
+        ]
+
+        return Future { $0(.success(productIdentifiers)) }
+            .delay(for: 0.5, scheduler: RunLoop.main)
```
Comment (leandroalonso) [inline Source/Services/InAppPurchasesService.swift]: Why do we need a delay?

**Review by leandroalonso — COMMENTED** [review]
@momo-ozawa I'm not being able to check the transaction on Xcode. It says "No Transactions" and I get this error on console: "Purchasing product is nil"

File: Source/Services/InAppPurchasesService.swift (hunk 1 above)
Comment (momo-ozawa) [reply Source/Services/InAppPurchasesService.swift]: Forgot to remove this -- fixed. 
2b499db

**Comment by momo-ozawa** [comment]
> @momo-ozawa I'm not being able to check the transaction on Xcode. It says "No Transactions" 

Hmm. I think this was happening because of framework linking issues.  b4a767e 

> I get this error on console: "Purchasing product is nil"

Fixed! a293585

**Review by leandroalonso — APPROVED** [review]

**Pull request merged**